### PR TITLE
feat: add create-transaction tool

### PR DESCRIPTION
**Pull request by trek-boldly-go**
Add new create-transaction tool that allows creating transactions in Actual Budget accounts. The tool supports:
- Required fields: account, date, amount
- Optional fields: payee, payee_name, category, notes, cleared, imported_id, transfer_id, imported_payee
- Subtransactions for split transactions
- Comprehensive input validation

Also includes:
- Full test suite with 19 test cases covering happy path, edge cases, and error handling
- Updated README.md to document the new tool
- createTransaction wrapper in actual-api.ts
- Tool registration in tools/index.ts

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/actual-api.ts [hunk 1]
```diff
@@ -232,3 +232,11 @@ export async function deleteCategoryGroup(id: string): Promise<unknown> {
   await initActualApi();
   return api.deleteCategoryGroup(id);
 }
+
+/**
+ * Create a transaction (ensures API is initialized)
+ */
+export async function createTransaction(accountId: string, args: Record<string, unknown>): Promise<string> {
```
Comment (s-stefanov) [inline src/actual-api.ts]: Can the input here be typed instead?

**Review by s-stefanov — CHANGES_REQUESTED** [review]

File: src/tools/create-transaction/index.ts [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+    required: ['account', 'date', 'amount'],
+  },
+};
+
+export async function handler(
+  args: Record<string, unknown>
+): Promise<ReturnType<typeof successWithJson> | ReturnType<typeof errorFromCatch>> {
+  try {
+    // Validate required fields
+    if (!args.account || typeof args.account !== 'string') {
```
[the remark this answers is not in the document]
Comment (trek-boldly-go) [reply src/tools/create-transaction/index.ts]: I missed the use of zod, will add in a new commit.

File: src/tools/create-transaction/index.ts [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+      return errorFromCatch('account is required and must be a string');
+    }
+
+    if (!args.date || typeof args.date !== 'string') {
+      return errorFromCatch('date is required and must be a string');
+    }
+
+    // Validate date format (YYYY-MM-DD)
+    if (!/^\d{4}-\d{2}-\d{2}$/.test(args.date)) {
```
[the remark this answers is not in the document]
Comment (trek-boldly-go) [reply src/tools/create-transaction/index.ts]: I missed the use of zod, will add in a new commit.

File: src/actual-api.ts (hunk 1 above)
Comment (trek-boldly-go) [reply src/actual-api.ts]: I was following the use of `Record<string, unknown>` I saw in a few of the other functions, but agree a typed arg is preferred. I will add this as well.
